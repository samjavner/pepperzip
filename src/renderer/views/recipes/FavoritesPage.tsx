import * as React from "react";

export const FavoritesPage: React.FunctionComponent = () => (
    <div>Favorites</div>
);

export default FavoritesPage;
