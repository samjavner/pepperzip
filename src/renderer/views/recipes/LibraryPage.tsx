import * as React from "react";

export const LibraryPage: React.FunctionComponent = () => <div>Library</div>;

export default LibraryPage;
