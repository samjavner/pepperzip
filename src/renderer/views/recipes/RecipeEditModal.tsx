import classNames from "classnames";
import * as React from "react";
import {
    isValid,
    Model,
    RecipeEdit,
    useRecipeEdit,
} from "../../core/recipes/recipeEdit";
import { useLocalStore } from "../../store/react";

const RecipeEditModalContainer: React.FunctionComponent<{
    title: string;
    init: Model;
    close: () => void;
    save: (model: Model) => void;
}> = ({ init, ...props }) => {
    const recipeEdit = useRecipeEdit(useLocalStore(), init);
    return <RecipeEditModal recipeEdit={recipeEdit} {...props} />;
};

export default RecipeEditModalContainer;

export const RecipeEditModal: React.FunctionComponent<{
    title: string;
    recipeEdit: RecipeEdit;
    close: () => void;
    save: (model: Model) => void;
}> = ({ title, recipeEdit, close, save }) => (
    <div
        className={classNames({
            modal: true,
            "is-active": true,
        })}
    >
        <div className="modal-background" />
        <div
            className="modal-card"
            style={{
                height: "calc(100% - 8rem)",
                width: "calc(100% - 8rem)",
            }}
        >
            <header className="modal-card-head">
                <p className="modal-card-title">{title}</p>
                <button className="delete" aria-label="close" onClick={close} />
            </header>
            <section className="modal-card-body">
                <div className="columns">
                    <div className="column">
                        <div className="field">
                            <label className="label">Name</label>
                            <div className="control">
                                <input
                                    className="input"
                                    type="text"
                                    value={recipeEdit.name}
                                    onChange={event =>
                                        recipeEdit.setName(event.target.value)
                                    }
                                />
                            </div>
                        </div>
                    </div>
                </div>
                <div className="columns">
                    <div className="column is-two-fifths">
                        <div className="field">
                            <label className="label">Ingredients</label>
                            <div className="control">
                                <textarea
                                    rows={Math.max(
                                        recipeEdit.ingredients.split("\n")
                                            .length,
                                        3
                                    )}
                                    className="textarea"
                                    value={recipeEdit.ingredients}
                                    onChange={event =>
                                        recipeEdit.setIngredients(
                                            event.target.value
                                        )
                                    }
                                />
                            </div>
                        </div>
                    </div>
                    <div className="column is-three-fifths">
                        <div className="field">
                            <label className="label">Directions</label>
                            <div className="control">
                                <textarea
                                    rows={Math.max(
                                        recipeEdit.directions.split("\n")
                                            .length,
                                        3
                                    )}
                                    className="textarea"
                                    value={recipeEdit.directions}
                                    onChange={event =>
                                        recipeEdit.setDirections(
                                            event.target.value
                                        )
                                    }
                                />
                            </div>
                        </div>
                    </div>
                </div>
            </section>
            <footer className="modal-card-foot">
                <button
                    className="button is-success"
                    disabled={!isValid(recipeEdit)}
                    onClick={() => save(recipeEdit)}
                >
                    Save Changes
                </button>
                <button className="button" onClick={close}>
                    Cancel
                </button>
            </footer>
        </div>
    </div>
);
