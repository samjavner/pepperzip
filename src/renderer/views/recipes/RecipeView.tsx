import {
    faCalendarPlus,
    faCartPlus,
    faCheck,
    faEdit,
    faEllipsisV,
    faHeart,
    faPrint,
    faStar,
    faThumbtack,
    faTrashAlt,
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import classNames from "classnames";
import * as React from "react";
import { Recipe } from "../../db/recipe";

export const RecipeView: React.FunctionComponent<{
    recipe: Recipe;
    onEdit: () => void;
}> = ({ recipe, onEdit }) => (
    <>
        <HeaderView recipe={recipe} onEdit={onEdit} />
        <TagsView recipe={recipe} />
        <ContentView recipe={recipe} />
    </>
);

const HeaderView: React.FunctionComponent<{
    recipe: Recipe;
    onEdit: () => void;
}> = ({ recipe, onEdit }) => (
    <div className="columns">
        <div className="column">
            <h1 className="title is-4">{recipe.name}</h1>
            <h2 className="subtitle is-6" style={{ marginBottom: 0 }}>
                <FontAwesomeIcon
                    fixedWidth={true}
                    icon={faStar}
                    className="has-text-primary"
                />
                <FontAwesomeIcon
                    fixedWidth={true}
                    icon={faStar}
                    className="has-text-primary"
                />
                <FontAwesomeIcon
                    fixedWidth={true}
                    icon={faStar}
                    className="has-text-primary"
                />
                <FontAwesomeIcon fixedWidth={true} icon={faStar} />
                <FontAwesomeIcon fixedWidth={true} icon={faStar} />
            </h2>
        </div>
        <div className="column is-narrow">
            <div className="field is-grouped">
                <div className="control">
                    <div className="field has-addons">
                        <div className="control">
                            <a className="button" href="#">
                                <span className="icon is-small">
                                    <FontAwesomeIcon icon={faCalendarPlus} />
                                </span>
                            </a>
                        </div>
                        <div className="control">
                            <a className="button" href="#">
                                <span className="icon is-small">
                                    <FontAwesomeIcon icon={faCartPlus} />
                                </span>
                            </a>
                        </div>
                    </div>
                </div>
                <div className="control">
                    <div className="field has-addons">
                        <div className="control">
                            <a className="button" href="#">
                                <span className="icon is-small has-text-info">
                                    <FontAwesomeIcon icon={faThumbtack} />
                                </span>
                            </a>
                        </div>
                        <div className="control">
                            <a className="button" href="#">
                                <span className="icon is-small has-text-success">
                                    <FontAwesomeIcon icon={faCheck} />
                                </span>
                            </a>
                        </div>
                        <div className="control">
                            <a className="button" href="#">
                                <span className="icon is-small has-text-danger">
                                    <FontAwesomeIcon icon={faHeart} />
                                </span>
                            </a>
                        </div>
                    </div>
                </div>
                <div className="dropdown is-right is-hoverable">
                    <div className="dropdown-trigger">
                        <button className="button">
                            <span className="icon is-small">
                                <FontAwesomeIcon icon={faEllipsisV} />
                            </span>
                        </button>
                    </div>
                    <div className="dropdown-menu">
                        <div className="dropdown-content">
                            <a
                                className="dropdown-item"
                                href="#"
                                onClick={onEdit}
                            >
                                <span className="icon is-small">
                                    <FontAwesomeIcon
                                        fixedWidth={true}
                                        icon={faEdit}
                                    />
                                </span>
                                <span
                                    style={{
                                        display: "inline-block",
                                        width: "0.5rem",
                                    }}
                                />
                                Edit
                            </a>
                            <a className="dropdown-item" href="#">
                                <span className="icon is-small">
                                    <FontAwesomeIcon
                                        fixedWidth={true}
                                        icon={faTrashAlt}
                                    />
                                </span>
                                <span
                                    style={{
                                        display: "inline-block",
                                        width: "0.5rem",
                                    }}
                                />
                                Delete
                            </a>
                            <hr className="dropdown-divider" />
                            <a className="dropdown-item" href="#">
                                <span className="icon is-small">
                                    <FontAwesomeIcon
                                        fixedWidth={true}
                                        icon={faPrint}
                                    />
                                </span>
                                <span
                                    style={{
                                        display: "inline-block",
                                        width: "0.5rem",
                                    }}
                                />
                                Print
                            </a>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
);

const TagsView: React.FunctionComponent<{
    recipe: Recipe;
}> = ({ recipe }) => (
    <div className="columns">
        <div className="column">
            <div className="field is-grouped is-grouped-multiline">
                {recipe.categories.map((category, index) => (
                    <div className="control">
                        <div className="tags">
                            <div className="tag is-dark" key={index}>
                                {category}
                            </div>
                        </div>
                    </div>
                ))}
                {recipe.servings && (
                    <div className="control">
                        <div className="tags has-addons">
                            <span className="tag is-dark">Servings:</span>
                            <span className="tag is-light">
                                {recipe.servings}
                            </span>
                        </div>
                    </div>
                )}
                {recipe.yield && (
                    <div className="control">
                        <div className="tags has-addons">
                            <span className="tag is-dark">Yield:</span>
                            <span className="tag is-light">{recipe.yield}</span>
                        </div>
                    </div>
                )}
                {recipe.source && (
                    <div className="control">
                        <div className="tags has-addons">
                            <span className="tag is-dark">Source:</span>
                            <span className="tag is-light">
                                {recipe.source}
                            </span>
                        </div>
                    </div>
                )}
                {recipe.author && (
                    <div className="control">
                        <div className="tags has-addons">
                            <span className="tag is-dark">Author:</span>
                            <span className="tag is-light">
                                {recipe.author}
                            </span>
                        </div>
                    </div>
                )}
                {recipe.webPage && (
                    <div className="control">
                        <div className="tags has-addons">
                            <span className="tag is-dark">Web Page:</span>
                            <span className="tag is-light">
                                {recipe.webPage}
                            </span>
                        </div>
                    </div>
                )}
                {recipe.sourcePageNumber && (
                    <div className="control">
                        <div className="tags has-addons">
                            <span className="tag is-dark">Page Number:</span>
                            <span className="tag is-light">
                                {recipe.sourcePageNumber}
                            </span>
                        </div>
                    </div>
                )}
                {recipe.copyright && (
                    <div className="control">
                        <div className="tags has-addons">
                            <span className="tag is-dark">Copyright:</span>
                            <span className="tag is-light">
                                {recipe.copyright}
                            </span>
                        </div>
                    </div>
                )}
                {recipe.publisher && (
                    <div className="control">
                        <div className="tags has-addons">
                            <span className="tag is-dark">Publisher:</span>
                            <span className="tag is-light">
                                {recipe.publisher}
                            </span>
                        </div>
                    </div>
                )}
                {recipe.publishDate && (
                    <div className="control">
                        <div className="tags has-addons">
                            <span className="tag is-dark">Publish Date:</span>
                            <span className="tag is-light">
                                {recipe.publishDate}
                            </span>
                        </div>
                    </div>
                )}
            </div>
        </div>
    </div>
);

const ContentView: React.FunctionComponent<{ recipe: Recipe }> = ({
    recipe,
}) => (
    <article className="message is-info" style={{ marginTop: "1.5rem" }}>
        <div className="message-body has-text-grey-dark">
            <div className="columns is-multiline">
                <div className="column is-two-fifths">
                    <ul>
                        {recipe.ingredients
                            .split("\n")
                            .map((ingredient, index) => (
                                <li
                                    key={index}
                                    className={classNames({
                                        "has-text-weight-bold": ingredient.startsWith(
                                            "# "
                                        ),
                                    })}
                                >
                                    {ingredient.startsWith("# ")
                                        ? ingredient.slice(2)
                                        : ingredient}
                                </li>
                            ))}
                    </ul>
                </div>
                <div className="column is-three-fifths">
                    <ul>
                        {recipe.directions
                            .split("\n")
                            .map((direction, index) => (
                                <li
                                    key={index}
                                    className={classNames({
                                        "has-text-weight-bold": direction.startsWith(
                                            "# "
                                        ),
                                    })}
                                    style={{ marginBottom: "1rem" }}
                                >
                                    {direction.startsWith("# ")
                                        ? direction.slice(2)
                                        : direction}
                                </li>
                            ))}
                    </ul>
                </div>
            </div>
        </div>
    </article>
);

export default RecipeView;
