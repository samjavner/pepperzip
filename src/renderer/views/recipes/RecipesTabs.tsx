import { Button, ButtonGroup } from "@blueprintjs/core";
import classNames from "classnames";
import * as React from "react";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import * as actions from "../../app/actions";
import { State } from "../../app/state";

function mapStateToProps(state: State) {
    return {
        activePage: state.activeRecipesPage,
    };
}

function mapDispatchToProps(dispatch: Dispatch<actions.Action>) {
    return {
        onLibraryClick: () => dispatch(actions.setActiveRecipesPage("library")),
        onRecipeBoxClick: () =>
            dispatch(actions.setActiveRecipesPage("recipe_box")),
        onFavoritesClick: () =>
            dispatch(actions.setActiveRecipesPage("favorites")),
        onTagsClick: () => dispatch(actions.setActiveRecipesPage("tags")),
        onSearchClick: () => dispatch(actions.setActiveRecipesPage("search")),
    };
}

type RecipesTabsProps = ReturnType<typeof mapStateToProps> &
    ReturnType<typeof mapDispatchToProps>;

export const RecipesTabs: React.SFC<RecipesTabsProps> = props => (
    <>
        <div className="pz-navtabs">
            <ButtonGroup minimal={true}>
                <Button
                    className={classNames({
                        "pz-navtabs-button": true,
                        "pz-navtabs-button--active":
                            props.activePage === "library",
                    })}
                    icon="book"
                    onClick={props.onLibraryClick}
                >
                    Library
                </Button>
                <Button
                    className={classNames({
                        "pz-navtabs-button": true,
                        "pz-navtabs-button--active":
                            props.activePage === "recipe_box",
                    })}
                    icon="box"
                    onClick={props.onRecipeBoxClick}
                >
                    Recipe Box
                </Button>
                <Button
                    className={classNames({
                        "pz-navtabs-button": true,
                        "pz-navtabs-button--active":
                            props.activePage === "favorites",
                    })}
                    icon="heart"
                    onClick={props.onFavoritesClick}
                >
                    Favorites
                </Button>
                <Button
                    className={classNames({
                        "pz-navtabs-button": true,
                        "pz-navtabs-button--active":
                            props.activePage === "tags",
                    })}
                    icon="tag"
                    onClick={props.onTagsClick}
                >
                    Tags
                </Button>
                <Button
                    className={classNames({
                        "pz-navtabs-button": true,
                        "pz-navtabs-button--active":
                            props.activePage === "search",
                    })}
                    icon="search"
                    onClick={props.onSearchClick}
                >
                    Search
                </Button>
            </ButtonGroup>
        </div>
        <div style={{ borderTop: "1px solid lightgray" }} />
    </>
);

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(RecipesTabs);
