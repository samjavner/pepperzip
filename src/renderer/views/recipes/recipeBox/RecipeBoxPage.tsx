import classNames from "classnames";
import * as React from "react";
import { connect } from "react-redux";
import {
    ArrowKeyStepper,
    AutoSizer,
    defaultCellRangeRenderer,
    Grid,
    GridCellRangeProps,
} from "react-virtualized";
import { Dispatch } from "redux";
import * as actions from "../../../app/actions";
import { State } from "../../../app/state";

function mapStateToProps(state: State) {
    return {
        selectedIndex: state.selectedRecipeIndex,
    };
}

function mapDispatchToProps(dispatch: Dispatch<actions.Action>) {
    return {
        setSelectedIndex: (index: number) =>
            dispatch(actions.setSelectedRecipeIndex(index)),
    };
}

/**
 * Implementation of cellRangeRenderer with caching disabled.
 * This is useful so that cells are re-rendered while stepping with arrow keys.
 */
function uncachedCellRangeRenderer(gridCellRangeProps: GridCellRangeProps) {
    return defaultCellRangeRenderer({
        ...gridCellRangeProps,
        cellCache: {},
    });
}

type RecipeBoxPageProps = ReturnType<typeof mapStateToProps> &
    ReturnType<typeof mapDispatchToProps>;

export const RecipeBoxPage: React.SFC<RecipeBoxPageProps> = props => (
    <div
        style={{
            height: "100%",
            display: "grid",
            gridTemplateColumns: "20rem 4fr",
        }}
    >
        <div style={{ gridColumn: 1 }}>
            <AutoSizer>
                {({ height, width }) => (
                    <ArrowKeyStepper
                        mode="cells"
                        isControlled={true}
                        columnCount={1}
                        rowCount={50000}
                        scrollToColumn={1}
                        scrollToRow={props.selectedIndex}
                        onScrollToChange={({ scrollToRow }) =>
                            props.setSelectedIndex(scrollToRow)
                        }
                    >
                        {({
                            onSectionRendered,
                            scrollToColumn,
                            scrollToRow,
                        }) => (
                            <Grid
                                onSectionRendered={onSectionRendered}
                                scrollToColumn={scrollToColumn}
                                scrollToRow={scrollToRow}
                                height={height}
                                width={width}
                                rowCount={50000}
                                columnCount={1}
                                rowHeight={25}
                                columnWidth={width}
                                autoContainerWidth={true}
                                cellRangeRenderer={uncachedCellRangeRenderer}
                                cellRenderer={({ key, rowIndex, style }) => (
                                    <div
                                        key={key}
                                        style={style}
                                        className={classNames({
                                            "has-background-link has-text-white":
                                                rowIndex ===
                                                props.selectedIndex,
                                        })}
                                        onClick={() =>
                                            props.setSelectedIndex(rowIndex)
                                        }
                                    >
                                        Recipe {rowIndex}
                                    </div>
                                )}
                                cell={true}
                            />
                        )}
                    </ArrowKeyStepper>
                )}
            </AutoSizer>
        </div>
        <div style={{ gridColumn: 2, overflowY: "auto" }}>
            <div>Index: {props.selectedIndex}</div>
            {[...Array(10).keys()].map(x => (
                <div className="section" key={x}>
                    Paragraph {x}
                </div>
            ))}
        </div>
    </div>
);

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(RecipeBoxPage);
