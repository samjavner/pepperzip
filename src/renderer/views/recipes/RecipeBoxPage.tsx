import { faPlus, faSearch } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import classNames from "classnames";
import * as React from "react";
import {
    ArrowKeyStepper,
    AutoSizer,
    defaultCellRangeRenderer,
    Grid,
    GridCellRangeProps,
} from "react-virtualized";
import * as uuid from "uuid";
import { RecipeBox } from "../../core/recipes/recipeBox";
import { Recipe } from "../../db/recipe";
import RecipeEditModal from "./RecipeEditModal";
import RecipeView from "./RecipeView";

/**
 * Implementation of cellRangeRenderer with caching disabled.
 * This is useful so that cells are re-rendered while stepping with arrow keys.
 */
function uncachedCellRangeRenderer(gridCellRangeProps: GridCellRangeProps) {
    return defaultCellRangeRenderer({
        ...gridCellRangeProps,
        cellCache: {},
    });
}

export const RecipeBoxPage: React.FunctionComponent<{
    recipeBox: RecipeBox;
}> = ({ recipeBox }) => {
    const recipe: Recipe | undefined =
        recipeBox.recipes[recipeBox.selectedIndex];

    return (
        <>
            <div
                style={{
                    height: "100%",
                    display: "grid",
                    gridTemplateColumns: "20rem 4fr",
                }}
            >
                <div
                    style={{
                        gridColumn: 1,
                        borderRight: "1px solid #c8cfd0", // $border
                        display: "grid",
                        gridTemplateRows: "auto 1fr",
                    }}
                >
                    <div
                        style={{
                            gridRow: 1,
                        }}
                    >
                        <div
                            className="field has-addons"
                            style={{ marginTop: "-1px" }}
                        >
                            <div className="control">
                                <input
                                    className="input"
                                    type="text"
                                    placeholder="Search recipes"
                                    style={{ width: "15rem", borderRadius: 0 }}
                                />
                            </div>
                            <div className="control">
                                <a
                                    className="button"
                                    href="#"
                                    style={{ borderRadius: 0 }}
                                >
                                    <FontAwesomeIcon icon={faSearch} />
                                </a>
                            </div>
                            <div className="control">
                                <a
                                    className="button"
                                    href="#"
                                    style={{ borderRadius: 0 }}
                                    onClick={recipeBox.openAddRecipe}
                                >
                                    <FontAwesomeIcon icon={faPlus} />
                                </a>
                            </div>
                        </div>
                    </div>
                    <div style={{ gridRow: 2, marginRight: "1px" }}>
                        <AutoSizer>
                            {({ height, width }) => (
                                <ArrowKeyStepper
                                    mode="cells"
                                    isControlled={true}
                                    columnCount={1}
                                    rowCount={recipeBox.recipes.length}
                                    scrollToColumn={1}
                                    scrollToRow={recipeBox.selectedIndex}
                                    onScrollToChange={({ scrollToRow }) =>
                                        recipeBox.setSelectedIndex(scrollToRow)
                                    }
                                >
                                    {({
                                        onSectionRendered,
                                        scrollToColumn,
                                        scrollToRow,
                                    }) => (
                                        <Grid
                                            onSectionRendered={
                                                onSectionRendered
                                            }
                                            scrollToColumn={scrollToColumn}
                                            scrollToRow={scrollToRow}
                                            height={height}
                                            width={width}
                                            rowCount={recipeBox.recipes.length}
                                            columnCount={1}
                                            rowHeight={25}
                                            columnWidth={width}
                                            autoContainerWidth={true}
                                            cellRangeRenderer={
                                                uncachedCellRangeRenderer
                                            }
                                            cellRenderer={({
                                                key,
                                                rowIndex,
                                                style,
                                            }) => (
                                                <div
                                                    key={key}
                                                    style={{
                                                        ...style,
                                                        overflow: "hidden",
                                                        textOverflow:
                                                            "ellipsis",
                                                        userSelect: "none",
                                                        whiteSpace: "nowrap",
                                                        width: "100%",
                                                    }}
                                                    className={classNames({
                                                        "has-background-link has-text-white is-selected-item":
                                                            rowIndex ===
                                                            recipeBox.selectedIndex,
                                                        "has-background-white":
                                                            rowIndex !==
                                                            recipeBox.selectedIndex,
                                                    })}
                                                    onClick={() =>
                                                        recipeBox.setSelectedIndex(
                                                            rowIndex
                                                        )
                                                    }
                                                >
                                                    {
                                                        recipeBox.recipes[
                                                            rowIndex
                                                        ].name
                                                    }
                                                </div>
                                            )}
                                            cell={true}
                                        />
                                    )}
                                </ArrowKeyStepper>
                            )}
                        </AutoSizer>
                    </div>
                </div>
                <div
                    style={{
                        gridColumn: 2,
                        overflowY: "scroll",
                        padding: "2rem",
                    }}
                >
                    {recipe && (
                        <RecipeView
                            recipe={recipe}
                            onEdit={recipeBox.openEditRecipe}
                        />
                    )}
                </div>
            </div>
            {recipeBox.isAddRecipeActive && (
                <RecipeEditModal
                    title="New Recipe"
                    init={{
                        id: uuid.v4(),
                        name: "New Recipe",
                        url: "",
                        description: "",
                        tags: [],
                        ingredients: "",
                        directions: "",
                        notes: "",
                        sourceText: "",
                        importWarnings: [],
                    }}
                    onSave={recipeBox.saveAddRecipe}
                    onClose={recipeBox.closeAddRecipe}
                />
            )}
            {recipeBox.isEditRecipeActive &&
                recipe && (
                    <RecipeEditModal
                        title="Edit Recipe"
                        init={recipe}
                        onSave={recipeBox.saveEditRecipe}
                        onClose={recipeBox.closeEditRecipe}
                    />
                )}
        </>
    );
};

export default RecipeBoxPage;
