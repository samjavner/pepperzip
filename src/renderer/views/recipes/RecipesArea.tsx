import * as React from "react";
import { connect } from "react-redux";
import { GlobalState } from "../../core/model";
import * as navigation from "../../core/navigation";
import FavoritesPage from "./favorites/FavoritesPage";
import LibraryPage from "./library/LibraryPage";
import RecipeBoxPage from "./recipeBox/RecipeBoxPage";
import RecipesTabs from "./RecipesTabs";
import SearchPage from "./search/SearchPage";
import TagsPage from "./tags/TagsPage";

interface RecipesAreaProps {
    activePage: navigation.RecipesPage;
}

const RecipesArea: React.SFC<RecipesAreaProps> = props => (
    <div
        style={{
            height: "100%",
            display: "grid",
            gridTemplateRows: "auto 1fr",
        }}
    >
        <div style={{ gridRow: 1 }}>
            <RecipesTabs />
        </div>
        <div style={{ gridRow: 2, overflow: "hidden" }}>
            {props.activePage === "library" && <LibraryPage />}
            {props.activePage === "recipe_box" && <RecipeBoxPage />}
            {props.activePage === "favorites" && <FavoritesPage />}
            {props.activePage === "tags" && <TagsPage />}
            {props.activePage === "search" && <SearchPage />}
        </div>
    </div>
);

const mapStateToProps = (state: GlobalState) => ({
    activePage: navigation.selectors.getActiveRecipesPage(state),
});

export default connect(mapStateToProps)(RecipesArea);
