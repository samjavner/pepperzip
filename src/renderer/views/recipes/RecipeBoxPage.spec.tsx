import * as React from "react";
import * as renderer from "react-test-renderer";
import * as recipeBox from "../../core/recipes/recipeBox";
import { RecipeBoxPage } from "./RecipeBoxPage";

describe("RecipeBoxPage", () => {
    it("renders the master/detail view", () => {
        const component = renderer.create(
            <RecipeBoxPage
                recipeBox={{
                    ...recipeBox.init,
                    selectedIndex: 10,
                    refreshSuccess: () => ({}),
                    setSelectedIndex: () => ({}),
                    openAddRecipe: () => ({}),
                    closeAddRecipe: () => ({}),
                    openEditRecipe: () => ({}),
                    closeEditRecipe: () => ({}),
                    refresh: () => Promise.resolve(),
                    saveAddRecipe: () => Promise.resolve(),
                    addMultiple: () => Promise.resolve(),
                }}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
