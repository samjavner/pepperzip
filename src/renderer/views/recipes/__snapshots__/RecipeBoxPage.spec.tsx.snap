// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeBoxPage renders the master/detail view 1`] = `
<div
  style={
    Object {
      "display": "grid",
      "gridTemplateColumns": "20rem 4fr",
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "borderRight": "1px solid #c8cfd0",
        "display": "grid",
        "gridColumn": 1,
        "gridTemplateRows": "auto 1fr",
      }
    }
  >
    <div
      style={
        Object {
          "gridRow": 1,
        }
      }
    >
      <div
        className="field has-addons"
        style={
          Object {
            "marginTop": "-1px",
          }
        }
      >
        <div
          className="control"
        >
          <input
            className="input"
            placeholder="Search recipes"
            style={
              Object {
                "borderRadius": 0,
                "width": "15rem",
              }
            }
            type="text"
          />
        </div>
        <div
          className="control"
        >
          <a
            className="button"
            href="#"
            style={
              Object {
                "borderRadius": 0,
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-search fa-w-16 "
              data-icon="search"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </div>
        <div
          className="control"
        >
          <a
            className="button"
            href="#"
            onClick={[Function]}
            style={
              Object {
                "borderRadius": 0,
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 "
              data-icon="plus"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "gridRow": 2,
          "marginRight": "1px",
        }
      }
    >
      <div
        style={
          Object {
            "height": 0,
            "overflow": "visible",
            "width": 0,
          }
        }
      >
        <div
          onKeyDown={[Function]}
        >
          <div
            aria-label="grid"
            aria-readonly={true}
            className="ReactVirtualized__Grid"
            onScroll={[Function]}
            role="grid"
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "boxSizing": "border-box",
                "direction": "ltr",
                "height": 0,
                "overflowX": "hidden",
                "overflowY": "hidden",
                "position": "relative",
                "width": 0,
                "willChange": "transform",
              }
            }
            tabIndex={0}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "gridColumn": 2,
        "overflowY": "auto",
        "padding": "2rem",
      }
    }
  />
</div>
`;
